azure-devops/provision-agent.ps1

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Sets up VM for use as a build machine
+
+Write-Output 'Starting...'
+
+if (Test-Path 'C:\agent') {
+    Write-Output 'Agent already installed, terminating.'
+    exit 0
+}
+
+Write-Output 'Agent doesn''t look installed yet'
+
+$AzureDevOpsURL = 'https://dev.azure.com/vclibs/'
+$AzureDevOpsPool = 'STL'
+[string]$PersonalAccessToken = $args[0]
+
+$WorkLoads =  '--add Microsoft.VisualStudio.Component.VC.CLI.Support ' + `
+              '--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 ' + `
+              '--add Microsoft.VisualStudio.Component.VC.Tools.ARM64 ' + `
+              '--add Microsoft.VisualStudio.Component.VC.Tools.ARM ' + `
+              '--add Microsoft.VisualStudio.Component.Windows10SDK.18362 '
+
+$ReleaseInPath = 'Preview'
+$Sku = 'Enterprise'
+$VSBootstrapperURL = 'https://aka.ms/vs/16/pre/vs_buildtools.exe'
+$CMakeURL = 'https://github.com/Kitware/CMake/releases/download/v3.16.4/cmake-3.16.4-win64-x64.msi'
+$LlvmURL = 'https://releases.llvm.org/9.0.0/LLVM-9.0.0-win64.exe'
+$NinjaURL = 'https://github.com/ninja-build/ninja/releases/download/v1.9.0/ninja-win.zip'
+$PythonURL = 'https://www.python.org/ftp/python/3.8.1/python-3.8.1-amd64.exe'
+$VstsAgentURL = 'https://vstsagentpackage.azureedge.net/agent/2.165.0/vsts-agent-win-x64-2.165.0.zip'
+
+$ErrorActionPreference = 'Stop'
+$ProgressPreference = 'SilentlyContinue'
+
+Function InstallVS
+{
+  Param(
+    [String]$WorkLoads,
+    [String]$Sku,
+    [String]$VSBootstrapperURL)
+  try
+  {
+    Write-Output 'Downloading VS bootstrapper ...'
+    [string]$bootstrapperExe = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName() + '.exe')
+    curl.exe -L -o $bootstrapperExe $VSBootstrapperURL
+
+    $Arguments = ('/c', $bootstrapperExe, $WorkLoads, '--quiet', '--norestart', '--wait', '--nocache')
+
+    Write-Output "Starting Install: $Arguments"
+    $proc = Start-Process -FilePath cmd.exe -ArgumentList $Arguments -Wait -PassThru
+    $exitCode = $proc.ExitCode
+
+    if ($exitCode -eq 0 -or $exitCode -eq 3010)
+    {
+        Write-Output "Installation successful! $exitCode"

```

Comment:
Should this say `Exit code: $exitCode` or something? `Installation successful! 3010` is the sort of bizarre message that I don't like to see in log files.